presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -220,6 +220,16 @@ else if (controlResult.getState() != State.SUCCESS) {
             }
         }
 
+        // Re-run the test query to confirm that the results don't match, in case the underlying data changed, but was cached on the test tier.
+        QueryResult results = executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), testTimeout, sessionProperties);
+        if (results.getState() != State.SUCCESS) {
+            return false;
+        }
+        testResult = results;
+        if (resultsMatch(controlResult, testResult)) {
+            return true;
+        }
+
```

Comment:
I guess it will be reasonable to run 3 additional times if the first result doesn't match. But the 3 additional runs must all match control.